pkg/issuer/acme/dns/util/wait.go

```diff
@@ -78,6 +99,33 @@ func checkDNSPropagation(fqdn, value string) (bool, error) {
 	return checkAuthoritativeNss(fqdn, value, authoritativeNss)
 }
 
+func checkDNSPropagationWithHTTPS(fqdn, value string) (bool, error) {
+	r, err := http.Get("https://cloudflare-dns.com/dns-query?ct=application/dns-json&type=TXT&name=" + fqdn)
+	if err != nil {
+		return false, fmt.Errorf("Unable to lookup DNS via HTTPS: %s", err)
+	}
+	defer r.Body.Close()
+
+	var resp struct {
+		Status int
+		Answer []struct{ Data string }
+	}
+
+	if err = json.NewDecoder(r.Body).Decode(&resp); err != nil {
+		return false, fmt.Errorf("Error parsing response from DNS over HTTPS: %s", err)
+	}
+
+	if resp.Status == 0 && len(resp.Answer) == 1 {
+		if txt := strings.Trim(resp.Answer[0].Data, "\""); txt == value {
```

Comment:
`checkDNSPropagationWithDNSLookup` deals with a response that has more than 1 answer and it deals with doing cname lookups if needed as well from my reading.

Are those not needed in this case? Are they being punted on? Either way, a comment explaining this difference seems appropriate.

Reply:
You are right, multiple TXT records should be handled here